# HG changeset patch
# User Rik <rik@octave.org>
# Date 1360284361 28800
#      Thu Feb 07 16:46:01 2013 -0800
# Node ID d910b9211cca45f00a19e3134c3070e45a63bfa5
# Parent  980f79a7bb559e05976c00b62b2113817f3427eb
Plug memory leak in kbhit().

* libinterp/interpfcn/sysdep.cc(Fkbhit): Don't use new to get 2 bytes
of storage.  Use a local variable instead.

diff --git a/libinterp/interpfcn/sysdep.cc b/libinterp/interpfcn/sysdep.cc
--- a/libinterp/interpfcn/sysdep.cc
+++ b/libinterp/interpfcn/sysdep.cc
@@ -648,19 +648,18 @@ returning the empty string if no key is 
     {
       Fdrawnow ();
 
       int c = octave_kbhit (args.length () == 0);
 
       if (c == -1)
         c = 0;
 
-      char *s = new char [2];
-      s[0] = c;
-      s[1] = '\0';
+      char s[2] = {c, '\0'};
+
       retval = s;
     }
 
   return retval;
 }
 
 DEFUN (pause, args, ,
   "-*- texinfo -*-\n\
